src/ray/gcs/call_back.h

```diff
@@ -0,0 +1,26 @@
+#ifndef RAY_GCS_CALL_BACK_H
+#define RAY_GCS_CALL_BACK_H
+
+#include <boost/optional/optional.hpp>
+#include <vector>
+#include "ray/common/status.h"
+
+namespace ray {
+
+namespace gcs {
+
+using StatusCallback = std::function<void(Status status)>;
+
+template <typename T>
+class DatumCallback {
+ public:
+  using SingleItem = std::function<void(Status status, boost::optional<T> datum)>;
+
+  using MultiItem = std::function<void(Status status, std::vector<T> datums)>;
+};
```

Comment:
These two callback types don't have to be defined in the same class? Just define 2 separate items.

Reply:
done